libraries/AP_Motors/AP_MotorsMulticopter.cpp

```diff
@@ -597,34 +607,39 @@ void AP_MotorsMulticopter::output_logic()
         limit.throttle_upper = true;
 
         // set and increment ramp variables
-        float spool_step = 1.0f / (_spool_up_time * _loop_rate);
         switch (_spool_desired) {
-        case DesiredSpoolState::SHUT_DOWN:
+        case DesiredSpoolState::SHUT_DOWN: {
+            const float spool_time = _spool_down_time > minimum_spool_time ? _spool_down_time : _spool_up_time;
+            const float spool_step = 1.0f / (spool_time * _loop_rate);
             _spin_up_ratio -= spool_step;
             // constrain ramp value and update mode
             if (_spin_up_ratio <= 0.0f) {
                 _spin_up_ratio = 0.0f;
                 _spool_state = SpoolState::SHUT_DOWN;
             }
             break;
+        }
 
-        case DesiredSpoolState::THROTTLE_UNLIMITED:
+        case DesiredSpoolState::THROTTLE_UNLIMITED: {
+            const float spool_step = 1.0f / (_spool_up_time * _loop_rate);
             _spin_up_ratio += spool_step;
             // constrain ramp value and update mode
             if (_spin_up_ratio >= 1.0f) {
                 _spin_up_ratio = 1.0f;
                 _spool_state = SpoolState::SPOOLING_UP;
             }
             break;
-
-        case DesiredSpoolState::GROUND_IDLE:
+        }
+        case DesiredSpoolState::GROUND_IDLE: {
+            const float spool_step = 1.0f / (_spool_up_time * _loop_rate);
```

Comment:
shouldn't this check the current state to see if its going up or down?

Reply:
Ah, okay, I wasn't tracking that state, I was mostly thinking about the complete on/off. I can add a check for that.

Reply:
Done. Please double check that though.